custom_components/bhyve/switch.py

```diff
@@ -515,6 +525,27 @@ def is_on(self):
         """Return the status of the sensor."""
         return self._is_on
 
+    async def set_smart_watering_soil_moisture(self, percentage):
+        """Set the soil moisture percentage for the zone."""
+        if self._attrs[ATTR_SMART_WATERING_ENABLED]:
+            landscape = await self._bhyve.get_landscape(self._device_id, self._zone_id)
+            if landscape is not None:
+                _LOGGER.debug("Landscape data %s", landscape)
+                update_json = {}
+                to_remove = ['available_water', 'created_at', 'distribution_uniformity', 'landscape_coefficient', 'plant_available_water', 'updated_at']
+
+                # Remove properties from update json
+                for item in landscape:
+                    if item not in to_remove:
+                        update_json[item] = landscape[item]
```

Comment:
It would be less brittle to construct a payload with known parameters which are required to update the landscape. If the API changes in the future it would be impossible to determine if a new parameter was added in the `get_landscape` response.